gen/service_test.go

```diff
@@ -665,11 +665,9 @@ func TestArgsAndResultValidation(t *testing.T) {
 			t.Fatalf("invalid test %q: either typ or serialize must be set", tt.desc)
 		}
 
-		x := reflect.New(typ)
-		args := []reflect.Value{reflect.ValueOf(tt.deserialize)}
-		e := x.MethodByName("FromWire").Call(args)[0].Interface()
-		if assert.NotNil(t, e, "%v: expected failure but got %v", tt.desc, x) {
-			assert.Contains(t, e.(error).Error(), tt.wantError, tt.desc)
+		x := reflect.New(typ).Interface().(serviceType)
```

Comment:
optional nit: I don't have enough background on the existing tests so this may be unnecessary, but you may want to use an `ok` check with `require` so if this test ever fails we don't see a panic.

(I see this three times in the second commit)

Reply:
Ibe more informative than something hand-written

Reply:
I considered that but the message from the runtime when that panics
will be more informative than the static string we can write there. (It
mentions which methods are missing, whether the method was on the
pointer rather than the value, etc.)

In this case, the panic has the same effect as failing the test for us:
catastrophic failure because we have a major bug in the codegen.

I would prefer to leave this as-is unless you feel strongly about this.

Reply:
sgtm 👍  just making sure it wasn't an oversight